File: android/app/src/main/java/com/dailymemory/domain/usecase/memory/GetRecentMemoriesUseCase.kt
package com.dailymemory.domain.usecase.memory

import com.dailymemory.domain.model.Memory
import com.dailymemory.domain.repository.MemoryRepository
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

/**
 * Use case for getting recent memories.
 */
class GetRecentMemoriesUseCase @Inject constructor(
    private val memoryRepository: MemoryRepository
) {
    operator fun invoke(limit: Int = 10): Flow<List<Memory>> {
        return memoryRepository.observeRecent(limit)
    }
}

File: ios/DailyMemory/Domain/UseCase/Memory/GetRecentMemoriesUseCase.swift
import Foundation

/// Use case for getting recent memories.
final class GetRecentMemoriesUseCase {
    private let memoryRepository: MemoryRepository

    init(memoryRepository: MemoryRepository = MemoryRepositoryImpl()) {
        self.memoryRepository = memoryRepository
    }

    func execute(limit: Int = 10) async throws -> [Memory] {
        return try await memoryRepository.getRecent(limit: limit)
    }
}
